src/main/groovy/de/felixschulze/gradle/HockeyAppUploadTask.groovy

```diff
@@ -102,23 +105,26 @@ class HockeyAppUploadTask extends DefaultTask {
             throw new IllegalArgumentException("Cannot upload to HockeyApp because API Token is missing")
         }
 
-        if (!applicationFile?.exists()) {
-            if (applicationFile) {
-                logger.debug("App file doesn't exist: "+applicationFile?.absolutePath)
-            }
+        if (!applicationFiles) {
             if (!applicationVariant && !hockeyApp.appFileNameRegex) {
                 throw new IllegalArgumentException("No appFileNameRegex provided.")
             }
             if (!hockeyApp.outputDirectory || !hockeyApp.outputDirectory.exists()) {
                 throw new IllegalArgumentException("The outputDirectory (" + hockeyApp.outputDirectory ? hockeyApp.outputDirectory.absolutePath : " not defined " + ") doesn't exists")
             }
-            applicationFile = FileHelper.getFile(hockeyApp.appFileNameRegex, hockeyApp.outputDirectory);
-            if (!applicationFile) {
+            applicationFiles = FileHelper.getFiles(hockeyApp.appFileNameRegex, hockeyApp.outputDirectory);
+            if (!applicationFiles) {
                 throw new IllegalStateException("No app file found in directory " + hockeyApp.outputDirectory.absolutePath)
             }
         }
 
-        logger.lifecycle("App file: " + applicationFile.absolutePath)
+        def appFilePaths = applicationFiles.collect { it.name }
+
+        if (!hockeyApp.allowMultipleAppFiles && applicationFiles.size() > 1) {
```

Comment:
Yea, I remember that. But unfortunately I don't know anything about that usages. Gives some other use cases multiple app files? Is there a way to get ordered applicationVariant.outputs or file list? The last could be OS/JRE specific, but I almost sure, that all existing implementations gives files unsorted. I mean, I believe that actually right now (almost) no one uses this plugin with multiple output files or multiple files, matching given regex.

For theoretically existing users, that after all have multiple outputs, you can increment major version number and declare, that plugin interface was changed.

Also the new major version could be a good point to start throwing an exception when getFile() actually found multiple files (with appropriate changes in HockeyAppUploadTask). And may be to set 'allowMultipleAppFiles' to false by default.

Of course It's your project and you should decide, whether to change the behavior or not. I made that changes for myself and just offer it to you.